src/main/java/me/iblitzkriegi/vixio/effects/message/EffEdit.java

```diff
@@ -0,0 +1,57 @@
+package me.iblitzkriegi.vixio.effects.message;
+
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import me.iblitzkriegi.vixio.Vixio;
+import me.iblitzkriegi.vixio.util.Util;
+import me.iblitzkriegi.vixio.util.wrapper.Bot;
+import net.dv8tion.jda.core.entities.Message;
+import org.bukkit.event.Event;
+
+public class EffEdit extends Effect {
+    static {
+        // No need for a bot since you can only edit your own messages
+        Vixio.getInstance().registerEffect(EffEdit.class, "edit %messages% to (show|say) %message/string%")
+                .setName("Edit Message")
+                .setDesc("A more natural way to edit a message. Don't forget that you can only edit your own messages.")
+                .setExample(
+                        "command /edit:",
+                        "\tprefixes: $",
+                        "\ttrigger:",
+                        "\t\treply with \"woah!\" and store it in {_message}",
+                        "\t\tedit {_message} to say \"this is cool!\""
+                );
+    }
+
+    private Expression<Message> messages;
+    private Expression<String> content;
+
+    @Override
+    protected void execute(Event e) {
+        Message content = Util.messageFrom(this.content.getSingle(e));
+        if (content == null) {
+            return;
+        }
+        for (Message message : messages.getAll(e)) {
+            Bot bot = Util.botFromID(message.getAuthor().getId());
```

Comment:
I have no idea what this is meant to be. You can't get a bot to be used from a id???

Reply:
you can only edit your own messages, so the author *has* to be one of the bots, or null